tests/mashupDocumentParser.test.ts

```diff
@@ -1,28 +1,85 @@
+import { TextDecoder, TextEncoder } from 'util';
 import MashupHandler from "../src/mashupDocumentParser";
 import { arrayUtils, pqUtils } from "../src/utils";
-import { simpleQueryMock, section1mSimpleQueryMock } from "./mocks";
+import { simpleQueryMock, section1mSimpleQueryMock, section1mNewQueryNameSimpleMock, section1mNewQueryNameBlankMock, relationshipInfo } from "./mocks";
 import base64 from "base64-js";
 import JSZip from "jszip";
 import WorkbookTemplate from "../src/workbookTemplate";
 import { section1mPath } from "../src/constants";
+import { Metadata } from "../src/types";
+
+(global as any).TextDecoder = TextDecoder;
+(global as any).TextEncoder = TextEncoder;
+
 describe("Mashup Document Parser tests", () => {
     test("ReplaceSingleQuery test", async () => {
         const mashupHandler = new MashupHandler();
 
         const defaultZipFile = await JSZip.loadAsync(WorkbookTemplate.SIMPLE_QUERY_WORKBOOK_TEMPLATE, { base64: true });
         const originalBase64Str = await pqUtils.getBase64(defaultZipFile);
 
-        const replacedQueryBase64Str = await mashupHandler.ReplaceSingleQuery(originalBase64Str, simpleQueryMock);
+          if (originalBase64Str) {
+            const replacedQueryBase64Str = await mashupHandler.ReplaceSingleQuery(originalBase64Str, "newQueryName", simpleQueryMock);
+            const buffer = base64.toByteArray(replacedQueryBase64Str).buffer;
+            const mashupArray = new arrayUtils.ArrayReader(buffer);
+            const startArray = mashupArray.getBytes(4);
+            const packageSize = mashupArray.getInt32();
+            const packageOPC = mashupArray.getBytes(packageSize);
+            const zip = await JSZip.loadAsync(packageOPC);
+            const section1m = await zip.file(section1mPath)?.async("text");
+            if (section1m) {
+                const mocksection1 = section1mNewQueryNameSimpleMock.replace(/ /g, "");
+                expect(section1m.replace(/ /g, "")).toEqual(mocksection1);
+            }
+          }
+    })
+
+    test("queryMetadataEntries test", async () => {
+        const defaultZipFile = await JSZip.loadAsync(WorkbookTemplate.SIMPLE_QUERY_WORKBOOK_TEMPLATE, { base64: true });
+        const originalBase64Str = await pqUtils.getBase64(defaultZipFile); 
+        if  (originalBase64Str) {
+            const handler = new MashupHandler() as any;
+            const { metadata } = handler.getPackageComponents(originalBase64Str);
+            const newMetadataArray = handler.editSingleQueryMetadata(metadata, {queryName: "newQueryName"});
+            const mashupArray = new arrayUtils.ArrayReader(newMetadataArray.buffer);
+            const metadataVersion = mashupArray.getBytes(4);
+            const metadataXmlSize = mashupArray.getInt32();
+            const metadataXml = mashupArray.getBytes(metadataXmlSize);
+
+            //parse metdataXml
+            const util= require('util');
```

Comment:
test seems to contain quite a bit of logic, perhaps here the best thing to do is compare with expected xml instead?

Reply:
You can still have 1 test that starts with the workbook template and covers the whole way, since logically we take a dependency on the template in our code, but for testing the details of the metadata xml - would be easier to test your logic in more scoped tests.

I have some ideas on how to simplify some of the code once it's in, having that separation and coverage would make the changes easier.